framework/behaviors/AttributeBehavior.php

```diff
@@ -80,6 +80,25 @@ class AttributeBehavior extends Behavior
      */
     public $value;
     /**
+     * @var array an array of the values indexed by attribute name that will be assigned to the current attributes respectively by attribute name.
+     * This property will be used only when [[value]] is `null`
+     * key - attribute name
+     * value - corresponding value
+     * Example:
+     * ```php
+     * 'values' => [
+     *   'number' => [$this, 'getNextNumber'],
+     *   'status' => 'updated'
+     * ]
+     * ```
+     * @see value for possible values
+     */
+    public $values = [];
+    /**
+     * @var string current processed attribute name. Can be use inside callable item of [[values]].
+     */
+    public $currentAttribute;
```

Comment:
May be discussed.
Fore external anonymous function.

And... I think new behavior (prev comment) is better idea.